packages/x-components/src/x-modules/recommendations/components/__tests__/recommendations.spec.ts

```diff
@@ -114,6 +114,27 @@ describe('testing recommendations component', () => {
 
     expect(isScopedSlotOverridden('custom-layout')).toBe(true);
   });
+
+  // eslint-disable-next-line max-len
+  it('renders at most the number of recommendations defined by `maxItemsToRender` prop', async () => {
+    const { wrapper, recommendations } = mountRecommendations();
+
+    const renderedRecommendations = (): WrapperArray<Vue> =>
+      findTestDataById(wrapper, 'recommendation-item');
+
+    await wrapper.setProps({ maxItemsToRender: 2 });
+    expect(renderedRecommendations()).toHaveLength(2);
+
+    await wrapper.setProps({ maxItemsToRender: 3 });
+    expect(renderedRecommendations()).toHaveLength(3);
+
+    await wrapper.setProps({ maxItemsToRender: 5 });
+    expect(renderedRecommendations()).toHaveLength(recommendations.length);
+  });
+
+  function findTestDataById(wrapper: Wrapper<Vue>, testDataId: string): WrapperArray<Vue> {
```

Comment:
I think you are overcomplicating things. Just create a method inside the `MountRecommendationsAPI` that retrieves the rendered suggestions. No need to create 2 separate method to do this just for aliasing the `findAll(getDataTestSelector` part.